language/move-lang/src/expansion/translate.rs

```diff
@@ -230,7 +231,19 @@ fn module(
     let alias_map = aliases(context, uses);
     context.set_and_shadow_aliases(alias_map.clone());
     let structs = structs(context, &name, pstructs);
-    let functions = functions(context, &name, pfunctions);
+    let (functions, spec_functions) = functions(context, &name, pfunctions);
+    let synthetics = psynthetics
+        .into_iter()
+        .map(|sp!(loc, s)| {
+            sp(
+                loc,
+                E::Synthetic_ {
+                    name: s.name,
+                    type_: single_type(context, s.type_),
```

Comment:
nit, if you bind `name` and `type_` as locals, and then use the local punning for instantiation, it should be a bit easier to read (should fit on one line)

Reply:
Done